### ISSUE TITLE
[base-ui][ModalManager] Select Option is not accessible when placed in a Modal or Drawer because aria-hidden is set

### ISSUE THREAD

**Issue by PMudra** [mui/base-ui#289]
### Steps to reproduce

Link to live example: https://stackblitz.com/edit/vitejs-vite-j2pkfu?file=src%2FApp.tsx

Example code:

```tsx
import { useState } from 'react';
import { Select, Option, Drawer, Button, Box } from '@mui/joy';

function App() {
  const [open, setOpen] = useState(false);

  return (
    <>
      <Button onClick={() => setOpen(true)}>Open</Button>

      <Drawer open={open} onClose={() => setOpen(false)}>
        <Box>
          <Select defaultValue="dog">
            <Option value="dog">Dog</Option>
            <Option value="cat">Cat</Option>
          </Select>
        </Box>
      </Drawer>
    </>
  );
}
```

Steps:
1. Press the "Open" button to open the Drawer.
2. Inspect the elements using dev tools.

### Current behavior

When opening the Drawer, the `ModalManager` will apply `aria-hidden="true"` to all elements in the root container (`<body>`) except for the _presentation_ container of the `Drawer` itself. Most of the time this is desired because only the the content of the Drawer should be accessible.

In my example though, I am nesting a `Select` component inside the `Drawer`. The list item menu of the `Select` (`Popper`) is placed below the `body` element and will have the `aria-hidden` attribute set to `true`:

![image](https://github.com/mui/material-ui/assets/84765/eb6ecd04-3fe4-4f08-bcd0-3a3d5dc821d5)

It seems that I am still able to use the `Select` component using the mouse and keyboard. So there seems to be no harm for the regular user, I think.

But we have noticed this bug in a playwright test using [`getByRole("option")`](https://playwright.dev/docs/api/class-page#page-get-by-role) which is strict about visibility and will not find the option because `aria-hidden` is set.

### Expected behavior

The options of the `Select` should be accessible when placed inside a modal/drawer.

Maybe:

1. `aria-hidden` should not be set to root-level containers that are part of the current modal.
2. Or: There should be some documentation on how to solve this issue otherwise.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: Linux 5.4 Ubuntu 20.04.6 LTS (Focal Fossa)
  Binaries:
    Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
    npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
    pnpm: 8.10.5 - ~/.local/share/pnpm/pnpm
  Browsers:
    Chrome: 123.0.6312.105
```
</details>

**Search keywords**: modal select aria-hidden modalmanager base-ui joy

**Comment by michaldudak** [comment]
@danilo-leal, this is about Joy's Select, so should remain in the Material UI repo.

---

@PMudra, You can render the listbox within the Select's DOM element by setting the `disablePortal` prop:

```tsx
<Select defaultValue="dog" slotProps={{ listbox: { disablePortal: true } }}>
```
